src/state/container/mod.rs

```diff
@@ -93,10 +109,11 @@ impl<'a> State<'a> {
     /// This is the recommended way to ensure the state
     /// is available in [Component::initialize](crate::framework::components::Component::initialize).
     #[track_caller]
-    pub fn require<T: CustomState<'a>>(&self) {
+    pub fn require<C, T: CustomState<'a>>(&self) {
         assert!(
             self.has::<T>(),
-            "operator requires {} state",
+            "{} requires {} state",
+            std::any::type_name::<C>(),
             std::any::type_name::<T>()
         );
     }
```

Comment:
It would be nice to have a usage example which illustrates the purpose of `C` and `T`.

Reply:
I added some more documentation and an example.